filter_ methods don't check type

```
In [2]: mt.filter_rows(5).count()
Java stack trace:
java.lang.AssertionError: assertion failed: bool, int32, IsNA(Ref(filterRows_pred,int32))
	at scala.Predef$.assert(Predef.scala:170)
	at is.hail.expr.ir.Infer$.apply(Infer.scala:37)
	at is.hail.expr.ir.Infer$.is$hail$expr$ir$Infer$$infer$1(Infer.scala:10)
	at is.hail.expr.ir.Infer$.apply(Infer.scala:42)
	at is.hail.expr.ir.Infer$.apply(Infer.scala:7)
```

--- a/python/hail/matrixtable.py
+++ b/python/hail/matrixtable.py
@@ -1295,7 +1295,7 @@ class MatrixTable(ExprContainer):
         warn("deprecation: 'drop_cols' will be removed before 0.2 release")
         return MatrixTable(self._jvds.dropCols())
 
-    @typecheck_method(expr=expr_any, keep=bool)
+    @typecheck_method(expr=expr_bool, keep=bool)
     def filter_rows(self, expr, keep: bool = True) -> 'MatrixTable':
         """Filter rows of the matrix.
 
@@ -1410,7 +1410,6 @@ class MatrixTable(ExprContainer):
         :class:`.MatrixTable`
             Filtered matrix table.
         """
-        expr = to_expr(expr)
         base, cleanup = self._process_joins(expr)
         analyze('MatrixTable.filter_cols', expr, self._col_indices, {self._row_axis})
 
@@ -1466,7 +1465,6 @@ class MatrixTable(ExprContainer):
         :class:`.MatrixTable`
             Filtered matrix table.
         """
-        expr = to_expr(expr)
         base, cleanup = self._process_joins(expr)
         analyze('MatrixTable.filter_entries', expr, self._entry_indices)
 

--- a/python/hail/table.py
+++ b/python/hail/table.py
@@ -635,7 +635,7 @@ class Table(ExprContainer):
             check_collisions(self._fields, k, self._row_indices)
         return self._select('Table.annotate', self.row.annotate(**named_exprs))
 
-    @typecheck_method(expr=anytype,
+    @typecheck_method(expr=expr_bool,
                       keep=bool)
     def filter(self, expr, keep=True):
         """Filter rows.
@@ -680,12 +680,8 @@ class Table(ExprContainer):
         :class:`.Table`
             Filtered table.
         """
-        expr = to_expr(expr)
         analyze('Table.filter', expr, self._row_indices)
         base, cleanup = self._process_joins(expr)
-        if expr.dtype != tbool:
-            raise TypeError("method 'filter' expects an expression of type 'bool', found '{}'"
-                            .format(expr.dtype))
 
         return cleanup(Table(base._jt.filter(expr._ast.to_hql(), keep)))
 

